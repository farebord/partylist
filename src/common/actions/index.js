export {
  ERROR_FETCHING_CURRENT_PLAYBACK,
  RECEIVED_CURRENT_PLAYBACK,
  REQUEST_CURRENT_PLAYBACK,
  SWITCH_PLAYING,
  fetchCurrentPlayback,
  switchPlaying,
} from './player';

export {
  RECEIVED_ACCESS_INFO,
  REQUEST_ACCESS_INFO,
  ERROR_FETCHING_ACCESS_INFO,
  fetchAccessInfo,
  fetchingAccessInfo,
} from './access';

export {
  RECEIVED_SEARCH_RESPONSE,
  SEARCHING,
  ERROR_SEARCHING,
} from './search';
