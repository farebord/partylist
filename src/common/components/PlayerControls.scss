%flex-centered {
  display: flex;
  align-items: center;
  justify-content: center;
}

.playerControl {
  @extend %flex-centered;
  flex: 1 0 auto;
}

.controls {
  flex: 0 0 auto;
  padding: 0 1rem;
};

.songInfo {
  @extend %flex-centered;
  flex-direction: column;
  flex: 1 0 auto;
}

.loadingProgress {
  margin: 0.5rem;
  color: white;
};

.songProgress {
  @extend %flex-centered;
  flex: 0 0 auto;
  padding-right: 1rem;
};