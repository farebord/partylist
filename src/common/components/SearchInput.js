import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SearchIcon from '@material-ui/icons/Search';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import { withStyles } from '@material-ui/styles';

const styles = () => ({
  root: {
    padding: '2px 4px',
    display: 'flex',
    alignItems: 'center',
  },
  input: {
    marginLeft: 8,
    flex: 1,
  },
  inputIcon: {
    padding: '0.5rem',
    margin: 0,
    display: 'flex',
    alignItems: 'center',
  },
});


export const SearchInput = ({ classes }) => (
  <Paper className={classes.root}>
    <InputBase
      className={classes.input}
      placeholder="Search..."
      inputProps={{ 'aria-label': 'Search...' }}
    />
    <span className={classes.inputIcon}>
      <SearchIcon />
    </span>
  </Paper>
);

SearchInput.propTypes = {
  classes: PropTypes.object.isRequired,
};

export const mapStateToProps = () => ({

});

export const mapDispatchToProps = () => ({

});

export default connect(null, mapDispatchToProps)(withStyles(styles)(SearchInput));
