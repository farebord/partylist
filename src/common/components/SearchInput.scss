%flex-centered {
  display: flex;
  align-items: center;
  justify-content: center;
}

.fieldContainer {
  @extend %flex-centered;
  padding: 0 0.25rem;
  flex: 1 0 auto;
  margin: 0 0.625rem;
}

.searchInput {
  margin-left: 0.5rem;
  flex: 1 1 auto;
}

.inputIcon {
  @extend %flex-centered;
  padding: 0.5rem;
  margin: 0;
}