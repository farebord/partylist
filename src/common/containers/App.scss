$default-padding-width: 0.625rem;

%flex-centered {
  display: flex;
  align-items: center;
  justify-content: center;
}

%flex-grow-only {
  flex: 1 0 50%;
}

%background-image-centered {
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
}

.wrapper {
  display: flex;
  height: 100vh;
  flex-direction: row;
  background-color: #f5f5f5
}

.searchContainer, .controlContainer, .tracksContainer {
  @extend %flex-centered;
  flex-direction: row;
  width: 100%;
  padding: $default-padding-width 0;
}

.playerContainer {
  @extend %flex-centered;
  @extend %flex-grow-only;
  flex-direction: column;
}

.middleContainer {
  @extend %flex-centered;
  @extend %background-image-centered;
  flex: 1 1 auto;
  width: 100%;
  box-sizing: border-box;
  border-left: inset $default-padding-width transparent;
  border-right: inset $default-padding-width transparent;
  background-size: contain;
}

.imageContainer{
  @extend %flex-centered;
  @extend %flex-grow-only;
  @extend %background-image-centered;
}

.tracksContainer {
  justify-content: flex-start;
  align-items: flex-start;
  flex: 1 1 auto;
  overflow-y: scroll;
  padding: 0;
}

@media (min-width: 1025px) {

  .tracksContainer {
    display: flex;
  }

  .middleContainer {
    display: none;
  }  
}

@media (max-width: 1024px) {
 
  .imageContainer {
    display: none !important;
  }

  .tracksContainer {
    display: none;
  }

  .middleContainer {
    display: flex;
  }
   
}