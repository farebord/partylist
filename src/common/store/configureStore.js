import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const configureStore = () => {
  const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(thunk)),
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      // eslint-disable-next-line global-require
      const nextRootReducer = require('../reducers').default;
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
};

export default configureStore;
