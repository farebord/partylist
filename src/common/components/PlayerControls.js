import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';


import { withStyles } from '@material-ui/styles';
import Grid from '@material-ui/core/Grid';
import Fab from '@material-ui/core/Fab';
import PlayIcon from '@material-ui/icons/PlayArrow';
import PauseIcon from '@material-ui/icons/Pause';
import CircularProgress from '@material-ui/core/CircularProgress';
import { switchPlaying } from '../actions';

const styles = () => ({
  controls: {
    flexGrow: '0',
    padding: '0 1rem',
  },
  songInfo: {

  },
  loadingProgress: {
    margin: '0.5rem',
    color: 'white',
  },
  songName: {

  },
  songArtists: {

  },
  songProgress: {
    paddingRight: '1rem',
  },
});

const renderButtonIcon = (loading, isPlaying, classes) => {
  if (loading) { return <CircularProgress size="small" className={classes.loadingProgress} />; }
  if (isPlaying) { return <PauseIcon />; }
  return <PlayIcon />;
};

const getArtists = artistList => artistList.map(artist => artist.name).join(', ');

/**
 * getTimeFromMs takes a number in ms and returns string in format MM:SS
 * @param {number} ms Number representing milliseconds
 * @return {string} Returns a string formated as MM:SS
 */
const getTimeFromMs = (ms) => {
  const minutes = Math.floor(ms / 60000);
  const seconds = ((ms % 60000) / 1000).toFixed(0);
  return `${minutes}:${seconds < 10 ? '0' : ''}${seconds}`;
};

export const getProgress = (progress, duration) => `${getTimeFromMs(progress)} / ${getTimeFromMs(duration)}`;

export const PlayerControls = ({
  progress, songInfo, classes, isPlaying, pauseResumePlayer, loading,
}) => (
  <React.Fragment>
    {songInfo && (
    <Grid container>
      <Grid item container className={classes.controls} xs justify="center" alignItems="center">
        <Fab size="small" color="secondary" aria-label="Add" onClick={pauseResumePlayer}>
          {renderButtonIcon(loading, isPlaying, classes)}
        </Fab>
      </Grid>
      <Grid container item direction="column" xs justify="center">
        <Grid item className={classes.songName}>{songInfo.name}</Grid>
        <Grid item className={classes.songArtists}>{getArtists(songInfo.artists)}</Grid>
      </Grid>
      <Grid item xs container alignItems="center" justify="flex-end" className={classes.songProgress}>
        {getProgress(progress, songInfo.duration_ms)}
      </Grid>
    </Grid>
    )}
  </React.Fragment>
);

PlayerControls.propTypes = {
  /* The progress of the song expressed in milliseconds */
  progress: PropTypes.number,
  /* Spotify Item object that contains all the information of a song */
  songInfo: PropTypes.object,
  /* Styles object */
  classes: PropTypes.object.isRequired,
  /* Boolean that says whether the song is being played or not */
  isPlaying: PropTypes.bool,
  /* Boolean that says if the pause/resume action is happening  */
  loading: PropTypes.bool,
  /* Function that toggles pause to the song being played. */
  pauseResumePlayer: PropTypes.func,
};

PlayerControls.defaultProps = {
  progress: null,
  loading: true,
  isPlaying: false,
  songInfo: null,
  pauseResumePlayer: () => {},
};

export const mapStateToProps = ({ player }) => ({
  progress: player.progress_ms,
  songInfo: player.item,
  isPlaying: player.is_playing,
  loading: player.playbackSwitching,
});

export const mapDispatchToProps = dispatch => ({
  pauseResumePlayer: () => dispatch(switchPlaying()),
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(PlayerControls));
