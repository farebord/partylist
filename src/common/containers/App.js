import React, { useEffect } from 'react'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/styles';

import { fetchCurrentPlayback, fetchAccessInfo } from '../actions'
import Grid from '@material-ui/core/Grid'
import SearchInput from '../components/SearchInput'
import PlayerControls from '../components/PlayerControls';

const styles = {
  containerGrid: {
    height: '100vh',
    padding: '10px'
  },
  searchBar: {
    flexGrow: '0'
  },
  playerControl: {
    flexGrow: '0'
  }
}

export const App = ({classes, getInitialData}) => {
  useEffect(() => {
    getInitialData()
  }, [])
  return (
    <Grid 
      container
      direction="column" 
      className={classes.containerGrid}
    >
      <Grid item xs className={classes.searchBar}>
        <SearchInput />
      </Grid>
      <Grid item xs>

      </Grid>
      <Grid item xs className={classes.playerControl}>
        <PlayerControls />
      </Grid>
    </Grid>
  )
}

App.propTypes = {
  classes: PropTypes.object,
  getInitialData: PropTypes.func,
  playerFetched: PropTypes.bool
}


export const mapStateToProps = () => ({
  
});

export const mapDispatchToProps = dispatch => ({
  getInitialData: () => {
    dispatch(fetchAccessInfo())
    dispatch(fetchCurrentPlayback())
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(App));
