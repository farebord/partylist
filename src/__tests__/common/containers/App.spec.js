import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme'
import { App, mapDispatchToProps } from 'common/containers/App';
import { Provider } from 'react-redux'
import configureStore from 'common/store/configureStore';
import SearchInput from 'common/components/SearchInput'
import PlayerControls from 'common/components/PlayerControls'

import { fetchAccessInfo, fetchCurrentPlayback } from 'common/actions'

jest.mock('common/actions')


describe('App component should', () => {
    it('render correctly', () => {
        const component = shallow(<App classes={{}} />)
        expect(component).toMatchSnapshot('App filters with no props');
    });

    it('render SearchInput', () => {
        const component = shallow(<App classes={{}} />)
        const searchInput = component.find(SearchInput)
        expect(searchInput).toHaveLength(1)
    })

    it('render PlayerControls', () => {
        const component = shallow(<App classes={{}} />)
        const playerControls = component.find(PlayerControls)
        expect(playerControls).toHaveLength(1)
    })

    it('call getAccessInfo when mounted', () => {
        const mock = jest.fn()
        let container = document.createElement('div');
        act(() => {
            ReactDOM.render(<Provider store={configureStore()}><App getAccessInfo={mock} classes={{}}/></Provider>, container)
        })
        expect(mock).toHaveBeenCalledTimes(1)  
    })
})

describe('App mapDispatchToProps should', () => {
    it('dispatch fetchAccessInfo when getAccessInfo its called', () => {
        const dispatch = jest.fn()
        fetchAccessInfo.mockReturnValueOnce({test: true})
        mapDispatchToProps(dispatch).getAccessInfo()
        expect(dispatch).toHaveBeenCalledTimes(1)
        expect(dispatch).toBeCalledWith({test: true})
    })

    it('dispatch fetchCurrentPlayback when getPlayingInfo its called', () => {
        const dispatch = jest.fn()
        fetchCurrentPlayback.mockReturnValueOnce({test: true})
        mapDispatchToProps(dispatch).getPlayingInfo()
        expect(dispatch).toHaveBeenCalledTimes(1)
        expect(dispatch).toBeCalledWith({test: true})
    })
})
