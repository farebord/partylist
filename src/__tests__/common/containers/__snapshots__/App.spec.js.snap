// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly: App filters with no props 1`] = `ShallowWrapper {}`;
